src/targets/node/fetch/client.ts

```diff
@@ -63,6 +64,13 @@ export const fetch: Client = {
           break;
         }
 
+        // The `form-data` module automatically adds a `Content-Type` header for `multipart/form-data` content and if we add our own here data won't be correctly transmitted.
+        // eslint-disable-next-line no-case-declarations -- We're only using `contentTypeHeader` within this block.
+        const contentTypeHeader = getHeaderName(headersObj, 'content-type');
+        if (contentTypeHeader) {
+          delete headersObj[contentTypeHeader];
```

Comment:
why can't we just remove the boundary then?  why do we need to remove the whole content-type header?

Reply:
Because `form-data` handles it already and if we add our own `Content-Type` header here the request ends up not sending the payload properly.

With the boundary:

```js
const fs = require('fs');
const FormData = require('form-data');
const fetch = require('node-fetch');
const formData = new FormData();
formData.append('foo', fs.createReadStream('hello.txt'));

let url = 'https://httpbin.org/post';

let options = {
  method: 'POST',
  headers: {'content-type': 'multipart/form-data; boundary=---011000010111000001101001'}
};

options.body = formData;

(async () => {
  await fetch(url, options)
    .then(res => res.json())
    .then(json => console.log(json))
    .catch(err => console.error('error:' + err));
})();
```

```js
{
  args: {},
  data: '',
  files: {},
  form: {},
  headers: {
    Accept: '*/*',
    'Accept-Encoding': 'gzip,deflate',
    'Content-Type': 'multipart/form-data; boundary=---011000010111000001101001',
    Host: 'httpbin.org',
    'Transfer-Encoding': 'chunked',
    'User-Agent': 'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)',
    'X-Amzn-Trace-Id': 'Root=1-62870691-559fcffb2d3a3e7d0322fd98'
  },
  json: null,
  origin: '98.45.207.127',
  url: 'https://httpbin.org/post'
}
```

Removing just the boundary:

```js
{
  args: {},
  data: '',
  files: {},
  form: {},
  headers: {
    Accept: '*/*',
    'Accept-Encoding': 'gzip,deflate',
    'Content-Type': 'multipart/form-data',
    Host: 'httpbin.org',
    'Transfer-Encoding': 'chunked',
    'User-Agent': 'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)',
    'X-Amzn-Trace-Id': 'Root=1-628706b0-5e47c68d1ced99ad1ca7a8a2'
  },
  json: null,
  origin: '98.45.207.127',
  url: 'https://httpbin.org/post'
}
```

Removing the `Content-Type` header completely:

```js
{
  args: {},
  data: '',
  files: { foo: 'hello world\n' },
  form: {},
  headers: {
    Accept: '*/*',
    'Accept-Encoding': 'gzip,deflate',
    'Content-Type': 'multipart/form-data;boundary=--------------------------753386540654200334809033',
    Host: 'httpbin.org',
    'Transfer-Encoding': 'chunked',
    'User-Agent': 'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)',
    'X-Amzn-Trace-Id': 'Root=1-628706c5-290386c3376e7ada57b332ca'
  },
  json: null,
  origin: '98.45.207.127',
  url: 'https://httpbin.org/post'
}
```

Reply:
this makes a lot more sense now that you explained it in such depth! thanks!